commandapi-core/src/main/java/dev/jorel/commandapi/AbstractCommandAPICommand.java

```diff
@@ -212,29 +247,69 @@ public void register() {
 			CommandAPI.logWarning("Command /" + meta.commandName + " is being registered after the server had loaded. Undefined behavior ahead!");
 		}
 		try {
-			Argument[] argumentsArr = (Argument[]) (args == null ? new AbstractArgument[0] : args.toArray(AbstractArgument[]::new));
+			Argument[] argumentsArray = (Argument[]) (arguments == null ? new AbstractArgument[0] : arguments.toArray(AbstractArgument[]::new));
 
 			// Check IGreedyArgument constraints
-			for (int i = 0, numGreedyArgs = 0; i < argumentsArr.length; i++) {
-				if (argumentsArr[i] instanceof IGreedyArgument) {
-					if (++numGreedyArgs > 1 || i != argumentsArr.length - 1) {
-						throw new GreedyArgumentException(argumentsArr);
+			for (int i = 0, numGreedyArgs = 0; i < argumentsArray.length; i++) {
+				if (argumentsArray[i] instanceof IGreedyArgument) {
+					if (++numGreedyArgs > 1 || i != argumentsArray.length - 1) {
+						throw new GreedyArgumentException(argumentsArray);
 					}
 				}
 			}
 
+			// Check optional argument constraints
+			// They can only be at the end, no required argument can follow an optional argument
+			int firstOptionalArgumentIndex = -1;
+			for (int i = 0, optionalArgumentIndex = -1; i < argumentsArray.length; i++) {
+				if (argumentsArray[i].isOptional()) {
+					firstOptionalArgumentIndex = i;
+					optionalArgumentIndex = i;
+					continue;
```

Comment:
This `continue` statement feels hacky. Your code flow goes:

```java
if (blah) {
  // stuff
}

if (not blah and other condition) {
  // other stuff
}
```

To catch your `not blah` condition, you've added a `continue` statement. This could be simplified using an `else` statement:

```java
if (blah) {
  // stuff
} else if (other condition) {
  // other stuff
}
```